### ISSUE TITLE
MokkeryGradleExtension incompatible version of Kotlin (2.1.0)

### ISSUE THREAD

**Issue by mgolebiowski95** [lupuuss/Mokkery#69]
Hi, after I upgrade kotlin version in my project to 2.1.0 I get the following error (convention plugin):

```
Class 'dev.mokkery.gradle.MokkeryGradleExtension' was compiled with an incompatible version of Kotlin. The actual metadata version is 2.1.0, but the compiler version 1.9.0 can read versions up to 2.0.0.
The class is loaded from /Users/maeb/.gradle/caches/modules-2/files-2.1/dev.mokkery/mokkery-gradle/2.6.1/2e9931366ad2ba742fb7addf6b2c052d1d854770/mokkery-gradle-2.6.1.jar!/dev/mokkery/gradle/MokkeryGradleExtension.class
```
and
```
Class 'dev.mokkery.MockMode' was compiled with an incompatible version of Kotlin. The actual metadata version is 2.1.0, but the compiler version 1.9.0 can read versions up to 2.0.0.
The class is loaded from /Users/maeb/.gradle/caches/modules-2/files-2.1/dev.mokkery/mokkery-core-jvm/2.6.1/7fb555d603e23d71d40d570a77d220ac145b0ff8/mokkery-core-jvm-2.6.1.jar!/dev/mokkery/MockMode.class
```

I use mokkery 2.6.1, which should be compatibile with kotlin 2.1.0. Gradle plugin was compile with different version?

**Comment by lupuuss** [comment]
Hi! 
**TL;DR From the error message it seems that Mokkery Gralde plugin being compiled with Kotlin 2.1.0 is actually the problem.  Bump Gradle to 8.12 (or maybe lower will be sufficient)**

You probably have this code in your convention plugin `build.gradle.kts`:
```kotlin
plugins {
    `kotlin-dsl`
}
```

Version of this plugin comes form the Gradle itself. [Each Gradle version has embedded Kotlin version](https://docs.gradle.org/current/userguide/compatibility.html). The Gradle version that you have uses Kotlin that understands  metadata up to 2.0.0 and Mokkery Gradle plugin was compiled with Kotlin 2.1.0. 

I was able to reproduce the issue with Gradle 8.5 and bumping to Gradle 8.12 solved it.
If you can't bump Gradle, you can also configure Mokkery without convention plugin.

**Comment by mgolebiowski95** [comment]
My entire project is based on convention plugins. It only happend with convention plugin, configuring mokkery without convention plugin works. To enable visibility of the mokkery extension, I added the **dev.mokkery:mokkery-gradle** dependency to the file, you indicated. I used AGP 8.7.3 with gradle 8.9 (gradle-wrapper.properties). Bump gradle to 8.12 fix the issue. Thanks for help.
